integrated_channels/integrated_channel/transmitters/learner_data.py

```diff
@@ -32,6 +32,100 @@ def __init__(self, enterprise_configuration, client=IntegratedChannelApiClient):
             client=client
         )
 
+    def single_learner_assessment_grade_transmit(self, exporter, **kwargs):
+        """
+        Send a single assessment level grade information to the integrated channel using the client.
+
+        Args:
+            exporter: The learner assessment data exporter used to send to the integrated channel.
+            kwargs: Contains integrated channel-specific information for customized transmission variables.
+                - app_label: The app label of the integrated channel for whom to store learner data records for.
+                - model_name: The name of the specific learner data record model to use.
+                - remote_user_id: The remote ID field name of the learner on the audit model.
+        """
+        TransmissionAudit = apps.get_model(  # pylint: disable=invalid-name
+            app_label=kwargs.get('app_label', 'integrated_channel'),
+            model_name=kwargs.get('model_name', 'LearnerDataTransmissionAudit'),
+        )
+        kwargs.update(
+            TransmissionAudit=TransmissionAudit,
+        )
+
+        # Even though we're transmitting a single learner, they can have records per assessment (multiple per course).
+        for learner_data in exporter.single_assessment_level_export(**kwargs):
+            serialized_payload = learner_data.serialize(enterprise_configuration=self.enterprise_configuration)
+            try:
+                code, body = self.client.create_course_completion(
+                    getattr(learner_data, kwargs.get('remote_user_id')),
+                    serialized_payload
+                )
+            except ClientError as client_error:
+                code = client_error.status_code
+                body = client_error.message
+                self.handle_transmission_error(learner_data, client_error)
+
+            learner_data.status = str(code)
+            learner_data.error_message = body if code >= 400 else ''
+
+            learner_data.save()
+
+    def assessment_level_transmit(self, exporter, **kwargs):
```

Comment:
hmm.. this model of passing the exporter itself is new? I thought we typically did `transmitter.transmit(exporter.export())` or something like it?

Reply:
learner data and content metadata are slightly different! 

**Course grade reporting**
```
    def transmit_learner_data(self, user):
        """
        Iterate over each learner data record and transmit it to the integrated channel.
        """
        exporter = self.get_learner_data_exporter(user)
        transmitter = self.get_learner_data_transmitter()
        transmitter.transmit(exporter)
```

**Assessment level grade reporting**
```
    def transmit_subsection_learner_data(self, user):
        """
        Iterate over each assessment learner data record and transmit them to the integrated channel.
        """
        exporter = self.get_learner_data_exporter(user)
        transmitter = self.get_learner_data_transmitter()
        transmitter.assessment_level_transmit(exporter)
```

**Content metadata transmission**
```
    def transmit_content_metadata(self, user):
        """
        Transmit content metadata to integrated channel.
        """
        exporter = self.get_content_metadata_exporter(user)
        transmitter = self.get_content_metadata_transmitter()
        transmitter.transmit(exporter.export())
```

Reply:
You can look at the base integrated channels models.py to see these.